vendor/github.com/mongodb/mongo-tools-common/util/exit_code.go

```diff
@@ -11,13 +11,23 @@ import (
 )
 
 const (
-	ExitError      int = 1
-	ExitClean      int = 0
-	ExitBadOptions int = 3
-	ExitKill       int = 4
-	// Go reserves exit code 2 for its own use
+	ExitFailure int = 1
+	ExitSuccess int = 0
 )
 
 var (
 	ErrTerminated = errors.New("received termination signal")
 )
+
+// SetupError is the error thrown by "New" functions used to convey what error occurred and the appropriate exit code.
+type SetupError struct {
+	Err error
+
+	// An optional message to be logged before exiting
+	Message string
```

Comment:
As mentioned above, maybe just capture the program name and *generate* the message instead.

Reply:
I tried to make this struct pretty generic in case we might find it useful in the future, but do you think it would be better if this was used only for indicating a usage message should be printed?

e.g.
```go
mf, err := mongofiles.New(...)
if err != nil {
    log.Logv(log.Always, err.Error())
    if _, ok := err.(SetupError); ok {
        log.Logv(log.Always, util.ShortUsage("mongofiles")
    }
    os.Exit(util.ExitFailure)
}
```